caikit/runtime/servicers/model_management_servicer.py

```diff
@@ -0,0 +1,94 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The Model Management Service is responsible for deploying and undeploying models
+"""
+
+# Third Party
+import grpc
+
+# First Party
+import alog
+
+# Local
+from caikit.interfaces.runtime.data_model import (
+    DeployModelRequest,
+    ModelInfo,
+    UndeployModelRequest,
+)
+from caikit.runtime.model_management.model_manager import ModelManager
+from caikit.runtime.types.caikit_runtime_exception import CaikitRuntimeException
+
+log = alog.use_channel("MM-SERVICR-I")
+
+
+# Define types for the proto versions of the DM classes
+DeployModelRequestProto = DeployModelRequest.get_proto_class()
+ModelInfoProto = ModelInfo.get_proto_class()
+UndeployModelRequestProto = UndeployModelRequest.get_proto_class()
+
+
+class ModelManagementServicerImpl:
+    __doc__ = __doc__
+
+    def __init__(self):
+        self._model_manager = ModelManager.get_instance()
+
+    def DeployModel(
+        self,
+        request: DeployModelRequestProto,  # type: ignore
```

Comment:
The need to type ignore here is sad, but I could not coerce the type checker to allow the return value of `get_proto_class` to be a valid type hint, no matter how much I tried to tell it what the class was.